src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -67,7 +67,18 @@ class wrap {
 
         auto units = co_await _os();
         auto guard = gate_guard(_g);
-        co_return co_await _h(std::move(rq), std::move(rp));
+        try {
+            co_return co_await _h(std::move(rq), std::move(rp));
+        } catch (kafka::client::partition_error const& ex) {
+            if (
+              ex.error == kafka::error_code::unknown_topic_or_partition
+              && ex.tp.topic == model::schema_registry_internal_tp.topic) {
+                throw exception(
+                  kafka::error_code::unknown_server_error,
+                  "_schemas topic does not exist");
```

Comment:
super nit: wouldn't this also throw if something went wrong with `_schemas` partitions? If so, should the error message also mention partition?

Reply:
I could just forward the underlying message `"{topic_partition: {_schemas}:0}, { error_code: unknown_topic_or_partition [3] }"}"`, but since it should only ever have 1 partition, and I'm not sure 0 partitions is a valid configuration, and Schema Registry is started (so the tp did exist), and it's probably had 5 retries over 5 seconds, the error message is probably close enough in the majority of instances.

Reply:
Works for me.